src/tls/Makefile

```diff
@@ -243,15 +243,37 @@ tls-lax: $(LIBFILES) ConcreteFlags $(FSTS)
 wc:
 	wc -l $(LIBFILES) $(FSTS)
 
+# ----------------------- FFI callback library -------------------------
+FFICallbacks.cmi: ../../libs/ml/FFICallbacks.mli
+	cp ../../libs/ml/FFICallbacks.mli .
+	ocamlfind c -package batteries -I $(FSTAR_HOME)/$(CONTRIB)/Platform/ml -I ../../libs -c FFICallbacks.mli
+
+FFICallbacks.cmo: FFICallbacks.cmi
+	
+FFICallbacks.cmx: FFICallbacks.cmi
+    
+FFICallbacks.cmx: ../../libs/ml/FFICallbacks.ml ffi.c FFICallbacks.cmi
+	cp ../../libs/ml/FFICallbacks.ml .
+	ocamlfind opt -package batteries $(OCAML_INCLUDE_PATHS) -c FFICallbacks.ml ffi.c
+    
+OCAMLMKLIB = ocamlfind mklib $(INCLUDE)    
+FFICallbacks.cmxa: FFICallbacks.cmx
```

Comment:
this seems correct, but I'm wondering if a recursive invocation of `make` in the `../../libs` directory might simplify things, rather than cluttering the main Makefile?